benchmark/transaction-log.bench.ts

```diff
@@ -0,0 +1,114 @@
+import { describe } from 'vitest';
+import { benchmark, concurrent } from './setup.js';
+
+describe('Transaction log', () => {
+	const ENTRY_COUNT = 1000;
+	const data = Buffer.alloc(100, 'a');
+
+	describe('write log with 100 byte records', () => {
+		benchmark('rocksdb', concurrent({
+			async setup(ctx) {
+				const db = ctx.db;
+				const log = db.useLog('0');
+				ctx.log = log;
+			},
+			async bench({ db, value, log, start, duration }) {
+				return db.transaction((txn) => {
+					log.addEntry(data, txn.id);
+				});
+			},
+		}));
+
+		benchmark('lmdb', concurrent({
+			async setup(ctx) {
+				let start = Date.now();
+				ctx.index = start;
+			},
+			bench(ctx) {
+				const { db } = ctx;
+				return db.put(ctx.index++, data) as unknown as Promise<void>;
+			},
+		}));
+	});
+	describe('read 100 iterators while write log with 100 byte records', () => {
+		benchmark('rocksdb', concurrent({
+			async setup(ctx) {
+				const db = ctx.db;
+				const log = db.useLog('0');
+				ctx.log = log;
+				ctx.iterators = Array(100).fill(null).map(() => log.query({ start: 1 }));
+			},
+			bench({ db, iterators, log }) {
+				let last;
+				for (let iterator of iterators) {
+					for (let entry of iterator) {
+						last = entry.timestamp;
+					}
+				}
+				return db.transaction((txn) => {
+					log.addEntry(data, txn.id);
+				}) as Promise<void>;
+			},
+			concurrencyMaximum: 2,
+		}));
+
+		benchmark('lmdb', concurrent({
+			async setup(ctx) {
+				let start = Date.now();
+				ctx.index = start;
+				ctx.last = 0;
+			},
+			bench(ctx) {
+				const { db, last } = ctx;
+				for (let i = 0; i < 100; i++) {
+					for (let { key, value } of db.getRange({ start: ctx.last })) {
+						ctx.last = key + 1;
+					}
+				}
+				return db.put(ctx.index++, data) as Promise<void>;
```

Comment:
Apparently lmdb-js' `put()` wants a string. Also need to force it to return a `Promise<void>`.

```suggestion
				return db.put(String(ctx.index++), data) as unknown as Promise<void>;
```